### PR TITLE
add train mode to model before bprop

### PR DESCRIPTION
**Pull request by rshih32**
if using cuda, the model must be in train mode for gradients to work

### CODE REVIEW

**Review by arridhana — COMMENTED** [review]
Do we also set this back to eval mode whenever we're done with computation that requires backprop?

**Comment by rshih32** [comment]
> Do we also set this back to eval mode whenever we're done with computation that requires backprop?

done

**Review by arridhana — APPROVED** [review]

**Comment by piotrm0** [comment]
You mean before fprop? fprop with train has non-determinism issues even on cpu.
